tools/markdown-to-html/docs-marked-renderer.spec.ts

```diff
@@ -0,0 +1,89 @@
+import {DocsMarkdownRenderer} from './docs-marked-renderer';
+
+describe('DocsMarkdownRenderer', () => {
+  let renderer: DocsMarkdownRenderer;
+  beforeEach(() => {
+    renderer = new DocsMarkdownRenderer();
+  });
+
+  it('generates regular headings for h1 and h2', () => {
+    expect(renderer.heading('a', 1, 'ignored')).toEqual('<h1>a</h1>');
+    expect(renderer.heading('b', 2, 'ignored')).toEqual('<h2>b</h2>');
+  });
+
+  it('creates header link for h3 and h4 headings', () => {
+    const heading3 = renderer.heading('heading text', 3, 'link-id');
+    expect(heading3).toEqual(`
+        <h3 id="link-id" class="docs-header-link">
+          <span header-link="link-id"></span>
+          heading text
+        </h3>
+      `);
+    const heading4 = renderer.heading('heading text', 4, 'second-link-id');
+    expect(heading4).toEqual(`
+        <h4 id="second-link-id" class="docs-header-link">
+          <span header-link="second-link-id"></span>
+          heading text
+        </h4>
+      `);
+  });
+
+  it('handles duplicate ids for headings', () => {
+    expect(renderer.heading('first', 3, 'id')).toContain('id="id"');
+    expect(renderer.heading('second', 3, 'id')).toContain('id="id-1"');
+  });
+
+  it('generates links', () => {
+    expect(renderer.link('something', 'some title', 'some text'))
+        .toEqual('<a href="guide/something" title="some title">some text</a>');
+    expect(renderer.link('guide/something', 'some title', 'some text'))
+        .toEqual('<a href="guide/something" title="some title">some text</a>');
+    expect(renderer.link('#some-hash', 'some title', 'some text'))
+        .toEqual('<a href="#some-hash" title="some title">some text</a>');
+    expect(renderer.link('http://google.com', 'some title', 'some text'))
+        .toEqual('<a href="http://google.com" title="some title">some text</a>');
+  });
+
+  it('generates html using new API', () => {
+    const result = renderer.html(`<!-- example(
+         {
+          "example": "exampleName",
+          "file": "example-html.html",
+          "region": "some-region",
```

Comment:
the final `,` messes up `JSON.parse` (one of the few things that was wrong with the new API 
 for DocsMarkedRenderer#html) 
I'll submit a patch to fix them

Reply:
👍 I'll wait for that fix before I submit this.